### PR TITLE
Replace deprecated Jaeger exporter

### LINKED ISSUE #3341: Replace deprecated `go.opentelemetry.io/otel/exporters/jaeger`
### What should be done?

https://pkg.go.dev/go.opentelemetry.io/otel/exporters/jaeger@v1.17.0 says:

> Deprecated: This module is no longer supported. OpenTelemetry dropped support for Jaeger exporter in July 2023. Jaeger officially accepts and recommends using OTLP. Use [go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp) or [go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc) instead.

We should replace that package.

### Where?

Currently, it is used only by our integration tests, but when this issue is picked up, it could be used elsewhere.

### Definition of Done

This package is no longer imported without the loss of functionality.

### PR DESCRIPTION
**Pull request by adetunjii**
# Description

Closes #3341.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [x] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [ ] I marked all done items in this checklist.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -103,9 +103,12 @@ services:
       context: ./build/deps
       dockerfile: jaeger.Dockerfile
     container_name: ferretdb_jaeger
+    environment:
+      - COLLECTOR_OTLP_ENABLED=true
     ports:
       - 16686:16686 # UI on http://127.0.0.1:16686/
       - 14268:14268 # to receive spans from clients
```
Comment (AlekSi) [inline docker-compose.yml:110]: Do we still need that port?

Comment (adetunjii) [reply docker-compose.yml:110]: We actually don't need 14268. I can remove it

**Review by rumyantseva — APPROVED** [review]
I tried everything locally, and the changes look good to me!
Thanks a lot for contributing!

**Review by AlekSi — APPROVED** [review]

**Comment by AlekSi** [comment]
Thank you!

**Pull request merged**